File: java/src/test/java/com/kcl/ListOptionsTest.java
package com.kcl;

import org.junit.Assert;
import org.junit.Test;

import com.kcl.api.API;
import com.kcl.api.Spec.ParseProgramArgs;
import com.kcl.api.Spec.ListOptionsResult;

public class ListOptionsTest {
    @Test
    public void testListOptionsApi() throws Exception {
        ParseProgramArgs args = ParseProgramArgs.newBuilder().addPaths("./src/test_data/option/main.k").build();

        API apiInstance = new API();
        ListOptionsResult result = apiInstance.listOptions(args);
        Assert.assertEquals(result.getOptions(0).getName(), "key1");
        Assert.assertEquals(result.getOptions(1).getName(), "key2");
        Assert.assertEquals(result.getOptions(2).getName(), "metadata-key");
    }
}

File: kotlin/src/test/kotlin/com/kcl/ListOptionsTest.kt
package com.kcl

import com.kcl.api.API
import com.kcl.api.parseProgramArgs
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals

class ListOptionsTest {
    companion object {
        private const val TEST_FILE = "./src/test_data/option/main.k"
    }

    @Test
    fun testListOptionsApi() {
        val args = parseProgramArgs { paths += TEST_FILE }
        val api = API()
        val result = api.listOptions(args)
        assertEquals("key1", result.optionsList[0].name)
        assertEquals("key2", result.optionsList[1].name)
        assertEquals("metadata-key", result.optionsList[2].name)
    }
}
